### ISSUE TITLE
Weird error message in misuse of addLayer()

### ISSUE THREAD

**Issue by stevage** [mapbox/mapbox-gl-js#10007]
Obviously not serious, but can be pretty confusing if you accidentally call `addLayer()` with the argument structure you expect from `addSource()`.

**mapbox-gl-js version**: 1.12.0

### Steps to Trigger Behavior

 1. call `map.addLayer('mylayerid', { ... })`

### Expected Behavior

An error message like "addLayer() expects an object, not a string."

### Actual Behavior

`Uncaught TypeError: cannot use 'in' operator to search for "ref" in "string"`

**Comment by mourner** [comment]
Agreed, even I sometimes do this mistake. Alternatively we could make this polymorphic so that `addLayer(id, obj)` works too (for API consistency) — not sure if there are any disadvantages.

**Comment by stevage** [comment]
Sounds even better. That inconsistency has always been a bit annoying. Both source and layer require an id, so there's no particular reason for the difference in parameter structure.
